### PR TITLE
Never store a squared root in SquaredDistance

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/20189)

### CODE REVIEW

File: components/style/values/animated/color.rs [hunk 1]
```diff
@@ -166,19 +166,19 @@ impl ComputeSquaredDistance for Color {
...
             Ok(
                 RGBA::transparent().compute_squared_distance(&other.color)? +
-                SquaredDistance::Value(1.),
+                SquaredDistance::from_sqrt(1.),
             )
         } else if self.is_numeric() && other.is_currentcolor() {
             Ok(
                 self.color.compute_squared_distance(&RGBA::transparent())? +
-                SquaredDistance::Value(1.),
+                SquaredDistance::from_sqrt(1.),
```
Comment (emilio) [inline components/style/values/animated/color.rs]: I really hope the compiler is able to figure these out, same for the `0.`.

File: components/style/properties/helpers/animated_properties.mako.rs [hunk 2]
```diff
@@ -811,7 +811,7 @@ impl Animate for Visibility {
 impl ComputeSquaredDistance for Visibility {
     #[inline]
     fn compute_squared_distance(&self, other: &Self) -> Result<SquaredDistance, ()> {
-        Ok(SquaredDistance::Value(if *self == *other { 0. } else { 1. }))
+        Ok(SquaredDistance::from_sqrt(if *self == *other { 0. } else { 1. }))
```
Comment (emilio) [inline components/style/properties/helpers/animated_properties.mako.rs]: Same, I'd really hope the compiler manages to inline all this and figure out that it doesn't actually need to do any multiplication. I'd say it's a safe bet but... :)

**Review by emilio — COMMENTED** [review]
This looks fine to me, but this means that now we sqrt unconditionally in https://searchfox.org/mozilla-central/rev/6cf5ac594a21db6df359dc49f4aa651fef59a27b/servo/ports/geckolib/glue.rs#456.

@hiikezoe, how important is that? IIUC the point of all this squared distance stuff is to avoid calling sqrt() when interpolating stuff that had multiple components, so I suspect it's fine, but want to check with you.

**Comment by hiikezoe** [comment]
It doen't look an important stuff to me either.   Currently computed distance is used only for showing timing function graph on animation inspector in devtools.  So it's not a big deal.   Also I don't quite understand why SquaredDistance originally has the two different values respectively.   nox introduced it in #18058 (right?), so I am fine if he is OK. :)

**Comment by emilio** [comment]
> It doen't look an important stuff to me either. Currently computed distance is used only for showing timing function graph on animation inspector in devtools. So it's not a big deal. Also I don't quite understand why SquaredDistance originally has the two different values respectively. nox introduced it in #18058 (right?), so I am fine if he is OK. :)

That was just to remove the separate `compute_distance` and `compute_squared_distance` functions, aiui.

**Comment by nox** [comment]
Yes, to keep the minor optimisation of not squaring and then computing the square root on properties that have very simple values. I don't think it matters to preserve that.

**Pull request merged**